lib/widgets/profile.dart

```diff
@@ -0,0 +1,261 @@
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+
+import '../api/model/model.dart';
+import '../model/content.dart';
+import '../model/narrow.dart';
+import 'content.dart';
+import 'message_list.dart';
+import 'page.dart';
+import 'store.dart';
+
+class ProfilePage extends StatelessWidget {
+  final int userId;
+
+  const ProfilePage({super.key, required this.userId});
+
+  static Route<void> buildRoute({required BuildContext context, required int userId}) {
+    return MaterialAccountWidgetRoute(context: context,
+      page: ProfilePage(userId: userId));
+  }
+
+  void _createDmNavigation (context) {
+    final store = PerAccountStoreWidget.of(context);
+    final narrow = DmNarrow.withUser(userId, selfUserId: store.account.userId);
+    Navigator.push(context,
+      MessageListPage.buildRoute(context: context, narrow: narrow));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final store = PerAccountStoreWidget.of(context);
+    final user = store.users[userId];
+    if (user == null) {
+      return const _ProfileErrorPage();
+    }
+
+    return Scaffold(
+      appBar: AppBar(title: Text(user.fullName)),
+      body: SingleChildScrollView(
+        child: Padding(
+          padding: const EdgeInsets.all(16),
+          child: DefaultTextStyle.merge(
+            softWrap: true,
+            textAlign: TextAlign.center,
+            style: const TextStyle(fontSize: 20, height: 1.5),
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
```

Comment:
I've applied the constraint here, but made it inside the scroll view so that the scroll bar is on the right-most part of the screen rather than on the right-most of the content